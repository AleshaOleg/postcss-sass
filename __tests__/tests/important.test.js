const getPostCssTreeFromSass = require('../../helpers/getPostCssTreeFromSassTree');

it('important.sass', function () {
    expect(getPostCssTreeFromSass('important')).toMatchSnapshot();
});
