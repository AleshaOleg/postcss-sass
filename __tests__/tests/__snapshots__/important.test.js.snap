// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`important.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "important": true,
          "prop": "color",
          "raws": Object {
            "before": "
    ",
            "between": ": ",
            "important": " !important",
            "semicolon": false,
          },
          "source": Object {
            "end": Object {
              "column": 25,
              "line": 2,
            },
            "input": Input {
              "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 2,
            },
          },
          "type": "decl",
          "value": "red",
        },
        Object {
          "nodes": Array [
            Object {
              "important": true,
              "prop": "image",
              "raws": Object {
                "before": "
        ",
                "between": ": ",
                "important": " !important",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 30,
                  "line": 4,
                },
                "input": Input {
                  "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 4,
                },
              },
              "type": "decl",
              "value": "none",
            },
            Object {
              "prop": "color",
              "raws": Object {
                "before": "
        ",
                "between": ": ",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 18,
                  "line": 5,
                },
                "input": Input {
                  "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 5,
                },
              },
              "type": "decl",
              "value": "red",
            },
          ],
          "raws": Object {
            "before": "
    ",
            "between": "",
          },
          "selector": "background:",
          "source": Object {
            "end": Object {
              "column": 19,
              "line": 5,
            },
            "input": Input {
              "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 1,
              "line": 3,
            },
          },
          "type": "rule",
        },
      ],
      "raws": Object {
        "before": "",
        "between": "",
      },
      "selector": ".important",
      "source": Object {
        "end": Object {
          "column": 19,
          "line": 5,
        },
        "input": Input {
          "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "type": "rule",
    },
  ],
  "raws": Object {
    "before": "",
    "semicolon": undefined,
  },
  "source": Object {
    "end": Object {
      "column": 19,
      "line": 5,
    },
    "input": Input {
      "css": ".important
    color: red !important
    background:
        image: none !important
        color: red
",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
